CenterForOpenScience/osf.io
# Fix permission classes for VOLs and Draft Registrations [PLAT-942] [PLAT-1078]

#### Purpose
Updates permission classes to handle view only link and draft registration objects. 

#### Changes
Retrieve node from view only link and draft registration objects so that `has_permission` is only called on objects that implement `has_permission`. 

#### Alternate Solution
Adding the following to the `DraftRegistration` model would also resolve the issue 
```
def has_permission(self, user, perms):
	return self.branched_from.has_permission(user, perms) 
```
but this method can't be implemented as simply for `PrivateLink` because of the `nodes` M2M field. 

#### QA Notes
_Not necessary because of tests_
Confirm draft registration detail and view only link detail endpoints do not 500 using the browsable api
* `/v2/nodes/<nid>/draft_registrations/<draft_id>/?format=api`
* `/v2/nodes/<nid>/view_only_links/<vol_id>/?format=api`

#### Ticket
https://openscience.atlassian.net/browse/PLAT-942
https://openscience.atlassian.net/browse/PLAT-1078
